### DISCUSSION TITLE
Flashed no problem - up and running no problem, but no intents are found. Where to start debug?

### DISCUSSION THREAD

**Discussion by dslugPX** [toverainc/willow#88] [Q&A] [answered]
This was no problem to get up and running following the readme. 

But I must be missing some step in Home Assistant that is likely nothing to do with using willow on the esp32 Box. 

Every command I try, including the most basic "Hi ESP turn off LightName" or "Hi ESP, turn off bedroom lights" responds with the correct text and No HA Intent found.

I had presumed from what I've read so far that the built in intents HassTurnOff $lightname would be supported once I was up and running.

Clearly there is something obvious I need to do in HA to make this go -  Can anyone point me towards what I'm looking for? Or where to start looking to debug?

**Comment by kristiankielhofner** [comment]
Glad to hear you got up and running with Willow!

In terms of the HA intents, your version of Home Assistant plays a big roll as that is where the intents are defined. The intent processing in 2023.5 is substantially better than earlier versions. We've put some effort into having compatibility with prior releases but we're learning more and more that for most people 2023.5 is highly recommended.

**Reply by dslugPX** [reply]
Thanks for the lightning fast response. 

I updated to latest between my post and your response too :). I was only one behind anyway - but I'm on 2023.5.3 now.

I can use the text based assistant in HA directly to type "turn on small signal" (small signal is a light I can see from my desk) and it works, but the exact same command using willow gives me a response that same No HA intent found. The command displayed on the box is the exact text I spoke, which is the exact text I typed in the assistant on HA. 

I recognize you are not focused on HA, but the broader arena of assistants (and I fully support that, and think the way you have articulated that fact - particularly on the reddit Home Assistant sub - was deftly handled as well, for what it's worth) - and I also recognize that I'm the worst kind of tester you want. Someone moderately capable, but not deeply well versed... But this stuff is cool and I want it to work!!! :) 

I really am presuming this is me not having HA set up properly, but I'm unsure where to start. If it's out bounds for you and here... I can jump over to reddit and get downvoted for asking questions, but might get an answer in the end! :) 

Cheers for all this work. I'm really excited to have this working, and glad I was able to grab something to test with before they sold out everywhere!

**Reply by kristiankielhofner** [reply]
We understood from the get-go that we'd end up getting pulled in to the world of HA so no worries :)!

This is truly strange and not something we've heard of/seen before, certainly with 2023.5. Question - does your text on the Willow display in a period at the end? That's the only time we've ever seen HA get confused by the command but it's a rare case and we haven't been able to reproduce it on 2023.5. Can you create an issue and include the full text of the serial monitor output? That will include the response from HA so we can have a better idea of what's going on here.

Thanks - we appreciate all of the early testing/feedback we're getting. Considering the soft announcement was a week ago today I think we're off to a good start!

**Reply by dslugPX** [reply]
Well wait one minute here - I've just seen: 

` WILLOW: sending '{"text":"turn on small signal.","language":"en"}' to Home Assistant API on 'http://homeassistant.local:8123/api/conversation/process'
E (03:17:59.021) esp-tls: couldn't get hostname for :homeassistant.local: getaddrinfo() returns 202, addrinfo=0x0
E (03:17:59.022) TRANSPORT_BASE: Failed to open a new connection: 32769
E (03:17:59.029) HTTP_CLIENT: Connection failed, sock < 0
E (03:17:59.035) WILLOW: failed to open HTTP connection: ESP_ERR_HTTP_CONNEC`

Well now.. Guess I should have looked a bit more through the logs before brining this up.

(but you did point me, perhaps unintentionally, towards where to start to see what the heck is up! thanks for that) 

I've got to get out of here for a bit - but that's a pretty telling error eh? 
Can't imagine why that's happening, but it's definitely on my end. I'll poke around.

**Reply by kristiankielhofner** [reply, marked as answer]
Hah, yeah that would do it! two things to check:

- Make sure the wifi network Willow is configured for can actually resolve homeassistant.local. Worst case scenario you can set the IP address or a hostname you know to work.
- You have TLS enabled. Make sure your Home Assistant instance running on port 8123 at the hostname/IP above is actually configured for TLS with cert, etc. By default it isn't.

**Reply by dslugPX** [reply]
This was all me, Fixed it right up when I saw that error. Vlan finickyness. 

This now "just works" and I'm over the moon. 

Super excited to actually dig into it, but I couldn't leave until I tried to sort it - so my dog is mad she's getting less park time today - but this is very very cool.

**Reply by kristiankielhofner** [reply]
It happens!

Glad to hear we got it sorted. I have to say it's also pretty flattering for someone to pick Willow time over man's best friend ;). Just make sure to get more park time tomorrow!

Speaking of tomorrow, we'll be releasing the Willow Inference Server so make sure to keep an eye out for that.

**Reply by dslugPX** [reply]
> Speaking of tomorrow, we'll be releasing the Willow Inference Server so make sure to keep an eye out for that.

I'm already looking at some of the GPU's you recommended on ebay!

I'm not really a well versed server person, despite having a ton of stuff running around here... it's mostly running on old macbooks or old mac minis, so I have to sort of figure out this bit of it. 

I've been in streaming media for as long as it's existed, but I was never a hardware person. Good news is I can follow tutorials pretty well so I'll get it sorted once I figure it out. Maybe will actually get a 1u rack mounted box and just move everything (Plex/Assorted Arrs/Home Assistant/MLBServer/etc) onto that and add WIS alongside all of them.

Be good to have something consolidated instead of this collection of frankenstein machines around the house... On the other hand the prospect of migrating everything is daunting... but that's my issue not yours :)  

I noticed you mention WSL, would going straight Ubuntu or other easy to use linux be smarter? I'd hate to have to go back to a world where windows updates were a thing that I had to think about. It's been a long time since I had to endure that pain and I'd like to continue to avoid it.

**Reply by kristiankielhofner** [reply]
If you go rack mount your best bet is the Tesla P4. If you get mini-tower-something the GTX 1070 is just about the best bang for the buck GPU on the market right now. They're both in the $90-$120 range depending on the usual used market fluctuations (different versions, sellers, etc).

I would go straight to Ubuntu - or potentially PopOS. It offers a great desktop GUI, is based on Ubuntu, and has a version that includes Nvidia drivers. This is very nice because typically Nvidia drivers are installed by the user and this is where the Nvidia driver horror stories you hear come from. With PopOS they're managed by PopOS and rendered invisible - "it just works". It's also one of the best Linux desktop experiences out there (IMO) for when/if you need a GUI. Then when you need to you can SSH into it (or whatever) and it's just Ubuntu.

With PopOS you can have WIS up and running in four commands (I'm going to distill it down to one).

When Willow is ready with easy-to-flash and configure firmware images the official "Getting Started" guide will likely recommend PopOS for users that are starting greenfield.

I only mention WSL because there are a lot of people with Nvidia-powered gaming systems that can install WIS in WSL today without buying anything.
